### DISCUSSION TITLE
ReceiveBeginPlay 如何正确触发?

### DISCUSSION THREAD

**Discussion by kimxee** [Tencent/puerts#1860] [Q&A] [answered]
UE的第三人称例子工程，TS继承了cpp的那个[项目名称]Character,然后工程的GameState指定为这个TS_Character。在TS_Character里按照例子里的,实现了ReceiveBeginPlay 函数，但是不能正确触发啊?
`
class TS_Player extends UE.NewGameCharacter{
    Constructor() {
        console.log("TS_Player Constructor called.");
    }
    ReceiveBeginPlay() : void
    {
        console.log("TS_Player ReceiveBeginPlay Begin");
    }
}
`
Constructor的打印会正常调用,ReceiveBeginPlay没反应。

**Comment by chexiongsheng** [comment]
我就没能碰到不触发的情况。

**Reply by kimxee** [reply, marked as answer]
可能是之前函数返回值没有表明void，不确定，现在能触发了

**Reply by chexiongsheng** [reply]
> 可能是之前函数返回值没有表明void，不确定，现在能触发了

ts没写返回值的意思是返回值为any
